File: myCTCA Mobile/iOS/src/myctca/Models/MyCTCAProxy.swift
//
//  MyCTCAProxy.swift
//  myctca
//
//  Created by Tomack, Barry on 2/22/18.
//  Copyright © 2018 CTCA. All rights reserved.
//

import Foundation

struct MyCTCAProxy: Codable {
    var toCTCAUniqueId: String = ""
    var relationshipType: String = ""
    var firstName: String = ""
    var lastName: String = ""
    var emailAddress: String?
    var expiresDateString: String?
    var permissions: [String]?
    var isImpersonating: Bool = false
    
    var fullName: String {
        get {
            return "\(firstName) \(lastName)"
        }
    }
    
    enum CodingKeys: String, CodingKey {
        case toCTCAUniqueId = "toCtcaUniqueId"
        case relationshipType, firstName, lastName, emailAddress, expiresDateString, permissions, isImpersonating
    }
    
    var expiresDate: Date {
        get {
            if let string = expiresDateString {
                let dateFormatter = DateFormatter()
                dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss"
                dateFormatter.locale = Locale(identifier: "en_US_POSIX")
                return dateFormatter.date(from: string)!
            } else {
                return Date()
            }
        }
    }

}

File: myCTCA Mobile/Android/src/myctca/app/src/main/java/com/myctca/model/MyCTCAProxy.java
package com.myctca.model;

/**
 * Created by tomackb on 2/23/18.
 */

public class MyCTCAProxy {

    private String toCtcaUniqueId;
    private String relationshipType;
    private String firstName;
    private String lastName;
    private boolean isImpersonating;

    public String getFullName() {
        return firstName + " " + lastName;
    }


    public String getToCtcaUniqueId() {
        return toCtcaUniqueId;
    }

    public String getRelationshipType() {
        return relationshipType;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public boolean isImpersonating() {
        return isImpersonating;
    }
}
